### PR TITLE
MDEXP-164 Mapping profile - transformations - process item material type

### PR DESCRIPTION
**Pull request by illia-borysenko**
## Purpose
To be able to use mapping profile transformation during generation marc file to replace key with values for a Item filed Material Type. 
## Approach
- Updated RuleFactory to add a new translation function for materialTypeId field
- Added a new client call to mod-inventory-storage to fetch reference data of material types
- Added new translation function to replace material type id with corresponding value using received reference data

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/org/folio/service/mapping/TranslationFunctionUnitTest.java [hunk 1]
```diff
@@ -139,6 +148,28 @@ void SetIdentifier_shouldReturnEmptyString_whenMetadataIsEmpty() {
...
+    TranslationFunction translationFunction = TranslationsHolder.lookup("set_material_type");
+    String value = "1a54b431-2e4f-452d-9cae-9cee66c9a892";
+    // when
+    String result = translationFunction.apply(value, 0, null, referenceData, null);
+    // then
+    Assert.assertEquals("book", result);
+  }
+
+  @Test
+  void SetMaterialType_shouldReturnEmptyString() {
```
Comment (andrii-novytskyi) [inline src/test/java/org/folio/service/mapping/TranslationFunctionUnitTest.java:163]: i think method name should start with lower case letter...but it is ok

Comment (illia-borysenko) [reply src/test/java/org/folio/service/mapping/TranslationFunctionUnitTest.java:163]: Let's make an exclusion for enumeration tests as far as other test cases have the same name convention, can discuss this point internally for the future to have the common convention

File: src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java [hunk 2]
```diff
@@ -82,6 +82,18 @@ public String apply(String locationId, int currentIndex, Translation translation
       }
     }
   },
+  SET_MATERIAL_TYPE() {
+    @Override
+    public String apply(String locationId, int currentIndex, Translation translation, ReferenceData referenceData, Metadata metadata) {
```
Comment (KVupp) [inline src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java:87]: locationId?

File: src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java [hunk 3]
```diff
@@ -82,6 +82,18 @@ public String apply(String locationId, int currentIndex, Translation translation
...
+      JsonObject entry = referenceData.getMaterialTypes().get(locationId);
+      if (entry == null) {
+        LOGGER.error("Location is not found by the given id: {}", locationId);
```
Comment (KVupp) [inline src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java:90]: Please correct the log message

File: src/test/resources/mapping/expected_marc_record_with_holdings_and_items.mrc [hunk 4]
```diff
@@ -0,0 +1 @@
...
  aitemLinkText2
  aitemLinkText3
  aitemLinkText4
1 aitemUri1
1 aitemUri2
1 aitemUri3
1 aitemUri4
  abook
  advd
ffic8b50e3f-0446-429c-960e-03774b88223f
```
Comment (KVupp) [inline src/test/resources/mapping/expected_marc_record_with_holdings_and_items.mrc:1]: Where is this file used? The corresponding file has been changed to a json in the master

File: src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java (hunk 2 above)
Comment (illia-borysenko) [reply src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java]: Updated

File: src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java (hunk 3 above)
Comment (illia-borysenko) [reply src/main/java/org/folio/service/mapping/processor/translations/TranslationsHolder.java]: Updated

File: src/test/resources/mapping/expected_marc_record_with_holdings_and_items.mrc (hunk 4 above)
Comment (illia-borysenko) [reply src/test/resources/mapping/expected_marc_record_with_holdings_and_items.mrc]: Removed

**Pull request merged**